### PR TITLE
refactor: do not pass private key to Sign() in linked data proofs

### PR DESCRIPTION
**Pull request by kdimak**
Signed-off-by: Dima <dkinoshenko@gmail.com>

#339

There will be additional PR to refactor the usage of private keys to build JWS from verifiable credential / presentation.

### CODE REVIEW

File: pkg/doc/signature/ed25519signature2018/suite.go [hunk 1]
```diff
@@ -56,34 +123,31 @@ func (s *SignatureSuite) GetDigest(doc []byte) []byte {
...
-	// ed25519 panics if key size is wrong
-	if l := len(pubKey); l != ed25519.PublicKeySize {
-		return errors.New("ed25519: bad public key length")
-	}
-
-	verified := ed25519.Verify(pubKey, doc, signature)
-	if !verified {
-		return errors.New("signature doesn't match")
+	if s.verifier == nil {
+		return ErrVerifierNotDefined
```
Comment (fqutishat) [inline pkg/doc/signature/ed25519signature2018/suite.go]: provide default sign

File: pkg/doc/signature/ed25519signature2018/suite.go [hunk 2]
```diff
@@ -56,34 +123,31 @@ func (s *SignatureSuite) GetDigest(doc []byte) []byte {
...
 	}
 
-	return nil
+	return s.verifier.Verify(pubKey, doc, signature)
```
Comment (fqutishat) [inline pkg/doc/signature/ed25519signature2018/suite.go]: remove withVerify

File: pkg/doc/signature/ed25519signature2018/suite.go [hunk 3]
```diff
@@ -21,16 +21,83 @@ import (
...
+// VerifierFunc is an adapter to allow the use of ordinary functions as suite verifier.
+type VerifierFunc func(pubKey, data, signature []byte) error
+
+// Verify will verify signature against public key.
+func (f VerifierFunc) Verify(pubKey, data, signature []byte) error {
+	return f(pubKey, data, signature)
+}
+
+// Ed25519Verifier is a ed25519 signature verifier.
+func Ed25519Verifier() VerifierFunc {
```
Comment (fqutishat) [inline pkg/doc/signature/ed25519signature2018/suite.go]: remove this

File: pkg/doc/signature/ed25519signature2018/suite.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/signature/ed25519signature2018/suite.go]: done

File: pkg/doc/signature/ed25519signature2018/suite.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/signature/ed25519signature2018/suite.go]: done

File: pkg/doc/signature/ed25519signature2018/suite.go (hunk 3 above)
Comment (kdimak) [reply pkg/doc/signature/ed25519signature2018/suite.go]: done

**Comment by kdimak** [comment]
Please do not close #339 . There will be extra PR to fix the marshaling of VC/VP to JWS.

File: pkg/doc/signature/ed25519signature2018/suite_test.go [hunk 4]
```diff
@@ -84,6 +92,44 @@ func TestSignatureSuite_Accept(t *testing.T) {
...
+func TestWithSigner(t *testing.T) {
+	suiteOpt := WithSigner(&mockSigner{})
+	require.NotNil(t, suiteOpt)
+
+	opts := &SignatureSuite{}
+	suiteOpt(opts)
+	require.NotNil(t, opts.signer)
+}
+
+/*func TestEd25519Verifier(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (kdimak) [reply pkg/doc/signature/ed25519signature2018/suite_test.go]: Ups, I will do it in the https://github.com/hyperledger/aries-framework-go/pull/1137.

**Pull request merged**